### PR TITLE
[Functions] Remove OCR Config Digest check

### PR DESCRIPTION
**Pull request by justinkaseman**

### CODE REVIEW

File: contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol [hunk 1]
```diff
@@ -142,27 +142,30 @@ contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilli
...
     ) {
-      revert ReportInvalid();
+      revert ReportInvalid(
+        "All transmissions on the report must be equal: requestIds, results, errors, onchainMetadata, offchainMetadata"
+      );
     }
 
     // Bounded by "MaxRequestBatchSize" on the Job's ReportingPluginConfig
-    for (uint256 i = 0; i < requestIds.length; ++i) {
+    for (uint256 i = 0; i < numberOfFulfillments; ++i) {
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol:160]: Let's not mix completely unrelated changes. Send out a separate PR for readability. Same for FunctionsBilling.sol.

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol [hunk 2]
```diff
@@ -19,9 +19,6 @@ abstract contract OCR2Base is ConfirmedOwner, OCR2Abstract {
     i_uniqueReports = uniqueReports;
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol:19]: We were planning to remove unique reports, right?

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol [hunk 3]
```diff
@@ -320,8 +316,10 @@ abstract contract OCR2Base is ConfirmedOwner, OCR2Abstract {
       emit Transmitted(configDigest, uint32(epochAndRound >> 8));
 
       ConfigInfo memory configInfo = s_configInfo;
-      // solhint-disable-next-line custom-errors
-      require(configInfo.latestConfigDigest == configDigest, "configDigest mismatch");
+      // The following check is disabled to allow both current and proposed routes to submit reports using the same OCR config digest
+      // Replay attacks are not a risk to Chanlink Functions because
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol:320]: Instead of calling out replay attacks you can make a more generic statement that request IDs are globally unique and validated in Coordinator.

Also typo "Chanlink".

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol [hunk 4]
```diff
@@ -320,8 +316,10 @@ abstract contract OCR2Base is ConfirmedOwner, OCR2Abstract {
       emit Transmitted(configDigest, uint32(epochAndRound >> 8));
 
       ConfigInfo memory configInfo = s_configInfo;
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol:318]: Do we need this line?

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol [hunk 5]
```diff
@@ -332,17 +330,12 @@ abstract contract OCR2Base is ConfirmedOwner, OCR2Abstract {
         expectedNumSignatures = configInfo.f + 1;
       }
 
-      // solhint-disable-next-line custom-errors
-      require(rs.length == expectedNumSignatures, "wrong number of signatures");
-      // solhint-disable-next-line custom-errors
-      require(rs.length == ss.length, "signatures out of registration");
+      if (rs.length != expectedNumSignatures) revert ReportInvalid("wrong number of signatures");
+      if (rs.length != ss.length) revert ReportInvalid("signatures out of registration");
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol:334]: This error message is confusing to me, what does it mean "out of registration" :) Maybe @stchrysa can help re-word.

**Review by bolekk — CHANGES_REQUESTED** [review]

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol (hunk 2 above)
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol:19]: Right

File: contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol (hunk 4 above)
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/v1_X/ocr/OCR2Base.sol]: It is used when determining the minimum signers, but I checked it with and without, it's cheaper gas-wise without.

File: contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol [hunk 6]
```diff
@@ -151,7 +151,9 @@ contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilli
       numberOfFulfillments != onchainMetadata.length ||
       numberOfFulfillments != offchainMetadata.length
     ) {
-      revert ReportInvalid();
+      revert ReportInvalid(
+        "All fields on the report must be equal: requestIds, results, errors, onchainMetadata, offchainMetadata"
```
Comment (bolekk) [inline contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol:155]: nit: "of equal length"

**Review by bolekk — APPROVED** [review]

File: contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol (hunk 6 above)
Comment (justinkaseman) [reply contracts/src/v0.8/functions/dev/v1_X/FunctionsCoordinator.sol]: Changed it

**Review by bolekk — APPROVED** [review]
